openpype/plugins/publish/integrate_new.py

```diff
@@ -1158,22 +1157,65 @@ def _add_alternative_sites(self,
         """
         conf_sites = system_sync_server_presets.get("sites", {})
 
+        alt_site_pairs = self._get_alt_site_pairs(conf_sites)
+
+        already_attached_keys = list(already_attached_sites.keys())
+        for added_site in already_attached_keys:
+            real_created = already_attached_sites[added_site]
+            for alt_site in alt_site_pairs.get(added_site, []):
+                if alt_site in already_attached_sites.keys():
+                    continue
+                meta = {"name": alt_site}
+                # alt site inherits state of 'created_dt'
+                if real_created:
+                    meta["created_dt"] = real_created
+                rec["sites"].append(meta)
+                already_attached_sites[meta["name"]] = real_created
+
+        return rec, already_attached_sites
+
+    def _get_alt_site_pairs(self, conf_sites):
+        """Returns dict of site and its alternative sites.
+
+        If `site` has alternative site, it means that alt_site has 'site' as
+        alternative site
+        Args:
+            conf_sites (dict)
+        Returns:
+            (dict): {'site': [alternative sites]...}
+        """
+        alt_site_pairs = {}
         for site_name, site_info in conf_sites.items():
             alt_sites = set(site_info.get("alternative_sites", []))
-            already_attached_keys = list(already_attached_sites.keys())
-            for added_site in already_attached_keys:
-                if added_site in alt_sites:
-                    if site_name in already_attached_keys:
+            if not alt_site_pairs.get(site_name):
+                alt_site_pairs[site_name] = []
+
+            alt_site_pairs[site_name].extend(alt_sites)
+
+            for alt_site in alt_sites:
+                if not alt_site_pairs.get(alt_site):
+                    alt_site_pairs[alt_site] = []
+                alt_site_pairs[alt_site].extend([site_name])
+
+        # transitive relationship, eg site is alternative to another which is
+        # alternative to nex site
+        loop = True
+        while loop:
+            loop = False
```

Comment:
What is this `loop` for? :D

Reply:
Or potentially easier: https://stackoverflow.com/a/41553897/1838864

Reply:
Or
```
for site_name, alt_sites in alt_site_pairs.items():
    sites_queue = collections.deque()
    sites_queue.extend(alt_sites)
    while sites_queue:
        alt_site = sites_queue.popleft()

        # safety against wrong config
        # {"SFTP": {"alternative_site": "SFTP"}
        if alt_site == site_name or alt_site not in alt_site_pairs:
            continue

        for alt_alt_site in alt_site_pairs[alt_site]:
            if (
                alt_alt_site != site_name
                and alt_alt_site not in alt_sites
            ):
                alt_sites.append(alt_alt_site)
                sites_queue.append(alt_alt_site)
```